src/main/java/ro/unibuc/hello/service/TeacherServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package ro.unibuc.hello.service;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import ro.unibuc.hello.dto.TeacherDto;
+import ro.unibuc.hello.models.TeacherEntity;
+import ro.unibuc.hello.repositories.TeacherRepository;
+
+import java.util.Optional;
+
+@Service
+public class TeacherServiceImpl implements TeacherService {
+
+    private final TeacherRepository teacherRepository;
+    private final ModelMapper modelMapper;
+
+    public TeacherServiceImpl(TeacherRepository teacherRepository,
+                              ModelMapper modelMapper) {
+        this.teacherRepository = teacherRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public TeacherEntity updateTeacherData(TeacherDto dto, String teacherId) {
+        Optional<TeacherEntity> teacher = teacherRepository.findById(teacherId);
```

Comment:
you should also check if dto is empty

Reply:
it is ok, if it is empty the entity remains the same.

Reply:
if we think about code execution, there wouldn't be any issues, but thinking about function logic, we should reconsider as it's not correctly the idea of returning the object not modified instead of an information about failed attempt to update

Reply:
ok, agreed